rules/REQUEST-932-APPLICATION-ATTACK-RCE.conf

```diff
@@ -359,6 +359,45 @@ SecRule REQUEST_COOKIES|!REQUEST_COOKIES:/__utm/|REQUEST_COOKIES_NAMES|ARGS_NAME
     setvar:'tx.inbound_anomaly_score_pl1=+%{tx.critical_anomaly_score}'"
 
 
+# [ Unix shell expressions - Bash Tilde expansion ]
+#
+# Detects the following patterns which are common in Unix shell scripts
+# and one-liners:
+#
+# ~+    $PWD
+# ~-    $OLDPWD
+# ~4    fifth directory from stack
```

Comment:
Do you know which shells support this feature? My bash takes it literally.

Reply:
Atleast Bash and Zsh. Try something like `echo ~+`. Zsh supports simply ~+, ~- etc.

Reply:
```
dune73@leander ~>bash --version
GNU bash, version 5.1.16(1)-release (x86_64-pc-linux-gnu)
Copyright (C) 2020 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>

This is free software; you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.
dune73@leander ~>echo ~4
~4

```

Reply:
For numbers to work directories must be pushed on to the stack. Kindly check the reference I've added.

Reply:
Ah, I got you. That's the push and pop stuff. I never used this but I see how this is coming in now.

OK, so this works. And sorry for not reading the documentation.